src/main/java/qna/service/QnaService.java

```diff
@@ -39,19 +39,19 @@ public void deleteQuestion(User loginUser, Long questionId) throws CannotDeleteE
             throw new CannotDeleteException("질문을 삭제할 권한이 없습니다.");
         }
 
-        List<Answer> answers = answerRepository.findByQuestionIdAndDeletedFalse(questionId);
+        List<Answer> answers = answerRepository.findByQuestionAndDeletedFalse(question);
         for (Answer answer : answers) {
             if (!answer.isOwner(loginUser)) {
                 throw new CannotDeleteException("다른 사람이 쓴 답변이 있어 삭제할 수 없습니다.");
             }
         }
 
         List<DeleteHistory> deleteHistories = new ArrayList<>();
-        question.setDeleted(true);
-        deleteHistories.add(new DeleteHistory(ContentType.QUESTION, questionId, question.getWriterId(), LocalDateTime.now()));
+        question.deleted();
+        deleteHistories.add(new DeleteHistory(ContentType.QUESTION, questionId, question.getWriter(), LocalDateTime.now()));
         for (Answer answer : answers) {
-            answer.setDeleted(true);
-            deleteHistories.add(new DeleteHistory(ContentType.ANSWER, answer.getId(), answer.getWriterId(), LocalDateTime.now()));
+            answer.deleted();
+            deleteHistories.add(new DeleteHistory(ContentType.ANSWER, answer.getId(), answer.getWriter(), LocalDateTime.now()));
```

Comment:
```suggestion
            deleteHistories.add(answer.toDeleteHistory());
```

데이터를 가진 객체에게 메시지를 전달한다면, `getter`에 대한 이슈를 줄일 수 있겠네요 :thinking:

Reply:
>"자율적인 객체들이 서로 메시지를 활용해 협력해 프로그래밍을 하는 것"이 객체지향 프로그래밍이라고 이야기한다. 객체는 섬이 아니고, 자율적이며, 스스로 구현을 결정할 수 있으며, 메시지를 통해 소통한다. 객체는 각자에게 가장 잘 맞는 책임과 역할을 부여하여 필요한 일을 해당 객체가 처리하도록 해야 한다.

Reply:
네 말씀주신 것처럼 message를 통해 도메인 모델에서 처리하도록 수정했습니다.